tests/dedicatedevhtests/l7_evh_crd_test.go

```diff
@@ -0,0 +1,440 @@
+	g.Expect(nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.JwtParams).To(gomega.BeNil())
+	g.Expect(nodes[0].SamlSpConfig).To(gomega.BeNil())
+
+	//Update with Oidc parameters as false
+	srUpdate := integrationtest.FakeSSORule{
+		Name:      srname,
+		Namespace: "default",
+		Fqdn:      "foo.com",
+		SSOType:   "OAuth",
+	}.SSORule()
+	srUpdate.ResourceVersion = "2"
+	oidcEnable, profile, userinfo := false, false, false
+	srUpdate.Spec.OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig = &v1alpha2.OIDCConfig{
+		OidcEnable: &oidcEnable,
+		Profile:    &profile,
+		Userinfo:   &userinfo,
+	}
+	_, err = v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Update(context.TODO(), srUpdate, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating SSORule: %v", err)
+	}
+	g.Eventually(func() string {
+		ssoRule, _ := v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Get(context.TODO(), srname, metav1.GetOptions{})
+		return ssoRule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+
+	g.Eventually(func() int {
+		_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		nodes, ok := aviModel.(*avinodes.AviObjectGraph)
+		if !ok {
+			return 0
+		}
+		return len(nodes.GetAviEvhVS())
+	}, 20*time.Second).Should(gomega.Equal(1))
+
+	// update is not getting reflected on evh nodes immediately. Hence adding a sleep of 5 seconds.
+	time.Sleep(5 * time.Second)
+
+	_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.OidcEnable).To(gomega.Equal(false))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Userinfo).To(gomega.Equal(false))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Profile).To(gomega.Equal(false))
+
+	// Delete/Disable
+	integrationtest.TeardownSSORule(t, g, sniVSKey, srname)
+	_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+
+	g.Expect(nodes[0].SsoPolicyRef).To(gomega.BeNil())
+	g.Expect(nodes[0].OauthVsConfig).To(gomega.BeNil())
+	g.Expect(nodes[0].SamlSpConfig).To(gomega.BeNil())
+
+	err = integrationtest.TearDownOAuthSecret()
+	if err != nil {
+		t.Fatalf("error in deleting my-oauth-secret: %v", err)
+	}
+	TearDownIngressForCacheSyncCheck(t, modelName)
+}
+
+func TestCreateUpdateDeleteSSORuleForEvhInsecure(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	modelName, _ := GetDedicatedModel("foo.com", "default")
+	srname := "samplesr-foo"
+	// create insecure ingress, SSORule should be applied in case of EVH
+	SetUpIngressForCacheSyncCheck(t, false, false, modelName)
+
+	err := integrationtest.SetUpOAuthSecret()
+	if err != nil {
+		t.Fatalf("error in creating my-oauth-secret: %v", err)
+	} else {
```

Comment:
`else` statement is not required. please move the code inside the else to outside.

Reply:
removed